jfix-stdlib-ratelimiter/src/test/kotlin/ru/fix/stdlib/ratelimiter/SuspendableRateLimitedDispatcherTest.kt

```diff
@@ -0,0 +1,390 @@
+package ru.fix.stdlib.ratelimiter
+
+import io.kotest.assertions.assertSoftly
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.matchers.booleans.shouldBeFalse
+import io.kotest.matchers.booleans.shouldBeTrue
+import io.kotest.matchers.collections.shouldContain
+import io.kotest.matchers.doubles.shouldBeBetween
+import io.kotest.matchers.longs.shouldBeInRange
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeInstanceOf
+import kotlinx.coroutines.*
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.future.future
+import mu.KLogging
+import org.awaitility.Awaitility.await
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.Timeout
+import ru.fix.aggregating.profiler.*
+import ru.fix.dynamic.property.api.DynamicProperty
+import java.lang.Thread.sleep
+import java.time.Duration
+import java.util.concurrent.*
+import java.util.concurrent.CompletableFuture.completedFuture
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+class SuspendableRateLimitedDispatcherTest {
```

Comment:
CompletableFuture is not the target API for which we producing this SuspendableRateLimitedDispatcher. That is why we need to focus our tests mostly on suspendable functions.